### ISSUE TITLE
Provide easy method/workflow to generate a map header

### ISSUE THREAD

**Issue by ehsteve** [sunpy/sunpy#2278]
Currently there is no good way for a user to know what should be or must be provided in a map header dictionary. The current process is cumbersome. This issue is to, at minimum, provide a convenience function to do this. At best we should provide a meta class object which solves all of our problems.

**Comment by Cadair** [comment]
I think the gen-0 version of this should just be a utility function (probably in `sunpy.utils`?) which outputs a `MetaDict` with the following keys (values provided for examples):
```
meta = MetaDict({
    'ctype1': 'HPLN-TAN',
    'ctype2': 'HPLT-TAN',
    'cunit1': 'arcsec', 'cunit2': 'arcsec',
    'crpix1': (data.shape[0] + 1)/2., 'crpix2': (data.shape[1] + 1)/2.,
    'cdelt1': 1.0, 'cdelt2': 1.0, 'crval1': 0.0, 'crval2': 0.0,
    'hgln_obs': 0.0,
    'hglt_obs': 0.0,
    'dsun_obs': const.au.to(u.m).value,
    'dsun_ref': const.au.to(u.m).value,
    'rsun_ref': const.R_sun.to(u.m).value,
    'rsun_obs': ((const.R_sun / const.au).decompose()
                 * u.radian).to(u.arcsec).value,
    't_obs': time_now.iso,
    'date-obs': time_now.iso,
})
```
it should be sufficient to pass this utility function, a SkyCoord for the observer from which hgln/lt_obs, dsun_obs, rsun_obs, t_obs, date-obs could be extracted. Then it should also take a Quantity for Cdelt and crval (defaulting to 0,0) and then the shape of the array and crpix should be optional defaulting to the centre of the array.

**Comment by Cadair** [comment]
@Alex-Ian-Hamilton has done a bit of research into required metadata as well here https://github.com/sunpy/sunpy/issues/1832#issuecomment-467987877

**Comment by Cadair** [comment]
I am going to deem this closed by #3083
